### ISSUE TITLE
Query Occasionally catch exception: "No more packets are available.: While executing Remote SQLSTATE: HY000."

### ISSUE THREAD

**Issue by zhaojintaozhao** [ByConity/ByConity#821]
## Bug Report
Query Occasionally catch exception: "No more packets are available.: While executing Remote SQLSTATE: HY000."

### Briefly describe the bug
Hi, All.
There are three servers and two default-workers in my ByConity cluster. 
I create a CnchMergeTree table and insert some values.
Then query sql from ByConity.

SomeTimes it returns the right result. 
SomeTimes it return the right result but catch exception: "No more packets are available.: While executing Remote SQLSTATE: HY000.". 

#### Step1:  create tabe.
```
CREATE TABLE test.t1
(`id` UInt32,
`pt_d` Date DEFAULT today()) 
ENGINE = CnchMergeTree()
PARTITION BY pt_d
Order by id
```
#### Step2: insert values
```
insert into test.t1 select (rand() % 1000), today() from numbers(50);
```
#### Step3: query sql
```
select id from test.t1 order by id limit 1;
```
SomeTimes it returns the right result. 
SomeTimes it return the right result but catch exception: "No more packets are available.: While executing Remote SQLSTATE: HY000.". 

#### server error stack: 
This is server error stack: 
```
<Error> executeQuery[370] : Code: 49, e.displayText() = DB::Exception: No more packets are available.: While executing Remote SQLSTATE: HY000 (version 21.8.7.1) (from xx.xx.xx.xx:xxx) (in query: select id from test.t1 order by id limit 1;), Stack trace (when copying this message, always include the lines below):

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ xxx in /opt/byconity/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ xxx in /opt/byconity/bin/clickhouse
2. DB::MultiplexedConnections::receivePacketUnlocked(std::__1::function<void (int, Poco::Timespan, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>) @ xxx in /opt/byconity/bin/clickhouse
3. DB::MultiplexedConnections::drain() @ xxx in /opt/byconity/bin/clickhouse
4. DB::RemoteQueryExecutor::finish(std::__1::unique_ptr<DB::RemoteQueryExecutorReadContext, std::__1::default_delete<DB::RemoteQueryExecutorReadContext> >*) @ xxx in /opt/byconity/bin/clickhouse
5. DB::RemoteSource::tryGenerate() @ xxx in /opt/byconity/bin/clickhouse
6. DB::ISource::work() @ xxx in /opt/byconity/bin/clickhouse
7. DB::SourceWithProgress::work() @ xxx in /opt/byconity/bin/clickhouse
8. ? @ xxx in /opt/byconity/bin/clickhouse
9. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ xxx in /opt/byconity/bin/clickhouse
10. DB::PipelineExecutor::executeSingleThread(unsigned long, unsigned long) @ xxx in /opt/byconity/bin/clickhouse
11. ? @ xxx in /opt/byconity/bin/clickhouse
12. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ xxx in /opt/byconity/bin/clickhouse
13. ? @ xxx in /opt/byconity/bin/clickhouse
```

#### worker error stack:
This is default-worker error stack:
```
<Error> Cnchworkerservice [200] : virtual void DB::CnchworkerserviceImpl::removeworkerResource(google::protobuf::Rpccontroller *, const Protos::RemoveWor kerResourceReq *, Protos::RemoveWorkerResourceResp *, google::protobuf::Closure *): Code: 372, e.displayText() = DB::Exception:Session not found. SQLSTATE: HYO00, Stack trace (when copying this message, always include the 1ines below) :

0. Poco::Exception::Exception (std::_1::basic_string<char, std::__1::char_traits<char>, std::_1::allocator<char> > const&, int) @ xxx in /opt/byconity/bin/clickhouse
1. DB::Exception::Exception (std::_1::basic_string<char, std::__1::char_traits<char>, std::_1::allocator<char> > const&, int, bool) @ xxx in /opt/byconity/bin/clickhouse
2. DB::NamedSessionsImp1<DB::Namedcnchsession>::acquiresession(unsigned long const&, std::_1::shared ptr<DB::Context const>, std::_1::chrono::duration<long long, std::_1::ratio<1l, 10000000001> >, boo1)-@ xxx in /opt/byconity/bin/clickhouse
3. DB::Context::acquireNamedCnchsession(unsigned long const&, std::_1::chrono::duration<long 1ong, std::_1::ratio<1l, 1 0000000001> >, bool) const @ xxx in /opt/byconity/bin/clickhouse
4. DB::CnchworkerserviceImpl::removeWorkerResource(google::protobuf:: RpcController*, DB::Protos:: RemoveWorkerResourceReq cOnst*, DB::Proto::RemoveWorkerResourceResp*. google::protobuf::Closure*) @ xxx- in /opt/byconity/bin/c1ickhouse
5. DB::Protos::CnchWorkerservice::CallMethod(google::protobuf::MethodDescriptor const*, google::protobuf::Rpccontroller*, google::protobuf::Message const*, google::protobuf::Message*, google::protobuf::Closure*) @ xxx in /opt/byconity/bin/clickhouse
6. brpc::policy::ProcessRpcRequest(brpc::InputMe5sageBase*) xxx in /opt/huawe i/app/bin/clickhouse
7. brpc::InputMessenger::OnNewMessages(brpc::Socket*) @ xxx in /opt/byconity/bin/clickhouse
8. brpc::socket:: ProcessEvent (void*) @ xxx in /opt/byconity/bin/clickhouse
9. bthread::TaskGroup::task_runner (long) @ xxx in /opt/byconity/bin/clickhouse 
```

### Version
0.2.0

Does any one face the same problem?

**Comment by zhaojintaozhao** [comment]
> I remembered seeing similar errors while on 0.2.0 but didn't notice it anymore on 0.2.2. Have you tried the newer version to see if it's fixed?

Thank you. I will update my system to 0.2.2

**Comment by TKaxe** [comment]
> ered seeing similar errors while on 0.2.0 but didn't notice it anymore on 0.2.2. Have you tried the newer version to see if it's fixed?

fix in #869
